src/main/java/io/jenkins/plugins/audit/UserCreationListener.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.audit;
+
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+
+import java.net.InetAddress;
+import java.time.Instant;
+
+import org.apache.logging.log4j.audit.event.CreateUser;
+import org.apache.logging.log4j.audit.LogEventFactory;
+import org.apache.logging.log4j.core.util.NetUtils;
+
+/**
+ * Listener notified of user account creation events.
+ */
+public abstract class UserCreationListener extends ExtensionPoint {
```

Comment:
On the right track here, but in reverse. We should either add an extension point to jenkins core so that we can implement it, or we should implement an extension point from jenkins core that provides notification of these events which we can translate into audit log events. For example, you could implement [SecurityListener](https://javadoc.jenkins.io/jenkins/security/SecurityListener.html) as an extension to get most of the info we need (if anything is missing, we might be able to propose an update to that class, or we'll have to add a new listener extension point similar to this one you proposed).